compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -750,32 +751,42 @@ fn transcribe_metavar_expr<'a>(
     Ok(())
 }
 
-/// Extracts an identifier that can be originated from a `$var:ident` variable or from a token tree.
-fn extract_ident<'a>(
+/// Extracts an metavariable symbol that can be an identifier, a token tree or a literal.
+fn extract_var_symbol<'a, 'interp>(
     dcx: DiagCtxtHandle<'a>,
     ident: Ident,
-    interp: &FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
-) -> PResult<'a, String> {
+    interp: &'interp FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
+) -> PResult<'a, &'interp Symbol> {
```

Comment:
```suggestion
) -> PResult<'a, Symbol> {
```
`Symbol` is just a `u32`, it is never passed by reference.

Reply:
Fixed.